onnxruntime/core/providers/cuda/tensor/split.cc

```diff
@@ -76,6 +76,31 @@ Status SplitKernel::ComputeInternal(OpKernelContext* ctx) const {
   auto input_dims = input_shape.GetDims();
   auto output_dimensions{input_shape.AsShapeVector()};
 
+  if (split_sizes.size() == 3 && (block_size_inside_axis_dim == 1)) {
```

Comment:
Split also supports data types like complex128 or string:https://onnx.ai/onnx/operators/onnx__Split.html
Please check that the data type is fixed size (not string) and size is not larger than 64 bit since your kernel does not support 128 bit.

Reply:
yes, the cpu ep can support complex128 and all other dtypes, but the cuda ep only supports FixedSizeTensorTypes. So here is ok to only support dtype size from 8bits to 64bits.